### PR TITLE
[FLINK-2368][ml]Adds convergence criteria

### PR DESCRIPTION
**Pull request by sachingoel0101**
Adds a convergence criteria class which allows the user to decide whether they want to terminate training at any point, based on the solutions before and after an iteration. 

Adding the first example to the SVM trainer.[Not finished yet].

### CODE REVIEW

File: flink-staging/flink-ml/src/main/scala/org/apache/flink/ml/classification/SVM.scala [hunk 1]
```diff
@@ -382,7 +402,11 @@ object SVM{
 
             // calculate the new weight vector by adding the weight vector updates to the weight
             // vector value
-            weights.union(weightedDeltaWs).reduce { _ + _ }
+            val newWeights = weights.union(weightedDeltaWs).reduce { _ + _ }
+            if(convergenceDefined){
+              converger.get.construct(input,weights,newWeights)
```
Comment (sachingoel0101) [inline flink-staging/flink-ml/src/main/scala/org/apache/flink/ml/classification/SVM.scala]: @tillrohrmann , could you take a look here? I am unable to work out the problem here.

**Comment by hsaputra** [comment]
Could you add [WIP] in the subject to help people to not review your PR now? Seemed like it is not ready?

**Comment by sachingoel0101** [comment]
@hsaputra , I don't understand. The title already has WIP in it. :confused:

**Comment by hsaputra** [comment]
Ah my bad, I missed it because usually I looked for it in the prefix. Never mind, sort about the confusion

**Comment by sachingoel0101** [comment]
Ah. Okay. No problem. :)

**Comment by zentol** [comment]
Closing since flink-ml is effectively frozen.

**Pull request closed without merging**